### PR TITLE
Move foreman-rails repo to foreman-release

### PR DESCRIPTION
**Pull request by ekohl**
This moves the foreman-rails repo to foreman-release. It's tied to the foreman release cycle and is versioned in the same way. This avoids a lot of the problems with ensuring we have the proper version installed and obsolete the older versions.

We also only depend on centos-release-scl-rh. We used to require SCLo repositories for Rails 4.2 but now we ship our own repository for this.

This is also intended to be backported to 1.17, 1.18 and 1.19. That's why I'd prefer it's merged before https://github.com/theforeman/foreman-packaging/pull/2822 because that's harder to backport.

### CODE REVIEW

**Comment by ekohl** [comment]
Will need https://github.com/theforeman/foreman-packaging/pull/2799 to build. After that's merged I'll rebase.

File: packages/foreman/foreman-release-scl/foreman-release-scl.spec [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
 License:  GPLv3+
-URL:      http://theforeman.org
-Source0:  tfm-ror51.repo
-Source1:  copr.gpg
+URL:      https://theforeman.org
 
 BuildArch: noarch
 
-Requires: centos-release-scl
+Requires: centos-release-scl-rh
```
Comment (ekohl) [inline packages/foreman/foreman-release-scl/foreman-release-scl.spec]: I'm leaning to doing this in a `7-2` so we can cherry pick it to 1.17 and 1.18.

File: packages/foreman/foreman/foreman.spec [hunk 2]
```diff
@@ -33,9 +33,12 @@ Source8: %{name}.gpg
 Source9: message_encryptor_extensions.rb
 Source10: %{executor_service_name}.sysconfig
 Source11: %{executor_service_name}.service
+Source12: %{name}-rails.repo
```
Comment (ekohl) [inline packages/foreman/foreman/foreman.spec]: Should this be a separate file or should we include it in `foreman.repo`?

**Comment by ekohl** [comment]
I'm not sure splitting out foreman-release actually makes it better and this is easier to cherry pick to older releases. It's green now so ready for a final review.

**Comment by ehelms** [comment]
Given the impending nature of 1.19, is it worth backporting to 1.17?

**Comment by ekohl** [comment]
We did talk about a 1.17.2 release and as it stands now that will introduce issues when we release packages so I do think it's worth it.

**Pull request merged**